keps/sig-api-machinery/555-server-side-apply/README.md

```diff
@@ -515,6 +516,19 @@ _This section must be completed when targeting beta graduation to a release._
       levels that could help debug the issue? The feature uses very little logging, and errors should be returned directly to the user.
       Not required until feature graduated to beta.
     - Testing: Are there any tests for failure mode? Failure modes are tested exhaustively both as unit-tests and as integration tests.
+-->
+  - SSA updates fail for pods with duplicated env. names or container ports.
+    - Known bug at least since 1.26
+    - Bugs:
+      - [Pod container ports and env-vars listMapKeys != validation](https://github.com/kubernetes/kubernetes/issues/113482)
+      - [Pod Garbage collector fails to clean up PODs from nodes that are not running anymore](https://github.com/kubernetes/kubernetes/issues/118261) (fixed in [#121103](https://github.com/kubernetes/kubernetes/pull/121103))
+    - Detection: The SSA request fails with 500. The response message is similar to the following:
+    `'failed to create manager for existing fields: failed to convert new object (app-b/app-b-5894548cb-7tssd; /v1, Kind=Pod) to smd typed: .spec.containers[name="app-b"].ports: duplicate entries for key [containerPort=8082,protocol="TCP"]'`.
+    - Mitigations: Make sure pods with duplicated keys for env. variables or
+      container pods are not created. Also, update the existing pods to cleanup
+      the problematic fields.
+    - Testing: [PodGC integration test](https://github.com/kubernetes/kubernetes/blob/7b9d244efd19f0d4cce4f46d1f34a6c7cff97b18/test/integration/podgc/podgc_test.go#L313)
+      reproduced the issue before withdrawing from SSA in PodGC in the [PR #121103](https://github.com/kubernetes/kubernetes/pull/121103).
```

Comment:
the fix for this is in flight in https://github.com/kubernetes-sigs/structured-merge-diff/pull/253, I don't think updating the design doc for this is helpful... the github issues mentioned seem sufficient, especially once the SSA impact is fixed

Reply:
I wasn't aware of the ongoing fix. 

Still, it might be worth it to inform users, who might be using older versions somehow, and the section "known failure modes" seems the valid place. However, if others think this is not the right place, then I'm happy to close it.

Reply:
I _think_ the "known failure modes" are about failure modes _inherent_ in the design, but will defer to @apelisse if he thinks it's useful to include this here

Reply:
I think this more of a decision of a PRR reviewer. What's your stance @wojtek-t?

Reply:
I think it might be useful to include it here, but since the fix is very close to land, I would wait to see how we manage to backport it and stuff before updating since we will have to update again soon anyways.

Reply:
Folks, so I have updated the note to only refer to pod status updates (which affected the pod failure policy), also mentioned that it is fixed in 1.29. :Let me know what you think

Reply:
I would personally merge it, but requires SIG lead approval.

@soltysh - can you please approve?

Reply:
> I would personally merge it, but requires SIG lead approval.
> 
> @soltysh - can you please approve?

This one falls under api-machinery so @deads2k, @jpbetz or @fedebongio

Reply:
Chasing the structured-merge-diff failure, it appears to have merged into kube in https://github.com/kubernetes/kubernetes/pull/121575.  Is this "SSA doesn't work under these conditions" still valid?

Reply:
IIUC the generic issue remains open: https://github.com/kubernetes/kubernetes/issues/113482
However, the https://github.com/kubernetes/kubernetes/pull/121575 PR fixes the use-case we had for pod disruption conditions, where subresource is patched.

Reply:
I have queued at item in the apimachinery agenda for "sometime after kep freeze": https://docs.google.com/document/d/1x9RNaaysyO0gXHIr1y50QFbiL1x8OWnk2v3XnrdkT5Y/edit